packages/next/shared/lib/router/router.ts

```diff
@@ -818,6 +818,7 @@ export default class Router implements BaseRouter {
     }
 
     let localeChange = options.locale !== this.locale
+    let prevLocale = this.locale
```

Comment:
```suggestion
    const prevLocale = this.locale
```
Let's change this to a const so we don't accidentally mutate it.

Reply:
Done 👍